Small bibtex changes

Chris1221 opened the pull request:
A few small things here:

- Fixed a typo in the Tennessen doi
- Writes genetic map citations to bibtex with a test
- Changes name of bibtex option to `bibtex-file` and adds the short `b` option

Chris1221 commented:
Whoops sorry, one moment, I overly mocked something.

jeromekelleher commented on stdpopsim/cli.py:
```diff
@@ -237,9 +237,12 @@ def write_bibtex(engine, model, contig, bibtex_file):
+    if contig.genetic_map is not None:
+        for citation in contig.genetic_map.citations:
+            bibtex_file.write(citation.fetch_bibtex())
+    if len(model.citations) > 0:
```
This check is redundant.

jeromekelleher commented on stdpopsim/cli.py:
```diff
@@ -237,9 +237,12 @@ def write_bibtex(engine, model, contig, bibtex_file):
+    if len(model.citations) > 0:
+        for citation in model.citations:
+            bibtex_file.write(citation.fetch_bibtex())
 
```
I forgot to check this --- are we closing the bibtex file after we finish writing to it? We could just put a ``bibtex_file.close()`` at the end of the method.

Chris1221 replied:
Right, sorry I had forgotten about this. I'll close it.

jeromekelleher commented on tests/test_cli.py:
```diff
@@ -463,6 +463,25 @@ def test_whole_bibex(self):
+
+        with mock.patch("builtins.open", mock.mock_open()):
+            with open('tmp.bib', 'w') as bib:
+                with mock.patch.object(
```
Why do we need two levels of patches here? Doesn't the first one catch them all?

Chris1221 replied:
My thought was that I wanted to check to ensure that specifically the genetic map `fetch_bibtex` was being called, not just the general one (which we're already testing). But since I don't think its possible to check if a specific instantiation of the class, I separately patched out the genetic map `fetch_bibtex` and ensured that it gets called (while the previous ones still need to be patched to avoid actually downloading anything). We could alternatively just patch out everything and ensure that it gets called the correct number of times, but I thought this was more explicit. Up to you!

jeromekelleher replied:
Ah, I see. I think asserting it gets called the expected number of times would be good, or else add in a comment describing the strategy. These mocking tests can be  a bit opaque at times.

Chris1221 commented:
So I removed the redundant check, closed the connection, and changed the test to be more transparent (removed the second mock, and simply asserted that `fetch_bibtex` was being called the right number of times). Rebased and squashed.

jeromekelleher commented on tests/test_cli.py:
```diff
@@ -463,6 +463,27 @@ def test_whole_bibex(self):
+                        stdpopsim.citations.Citation,
+                        "fetch_bibtex") as mock_bib:
+                    cli.write_bibtex(engine, model, contig, bib)
+                    assert mock_bib.call_count == ncite
```
This should be ``self.assertEqual()`` as we're using nose rather than pytest.

jeromekelleher commented:
Looks great, thanks @Chris1221. One minor issue above.

Chris1221 commented:
Whoops, my bad @jeromekelleher. Should be fixed now.
